### PR TITLE
feat(vis): align visual appearance, rotate bar plot axis labels, add vis hooks, remove raincloud plot

### PR DESCRIPTION
**Pull request by dvmoritzschoefl**
Innovation day from friday with @dvdanielamoitzi 
- Fixes a few hexbin vis issues and labels
- Deleted raincloud plot
- Aligned axis and tick label colors and sizes for all plots
- Aligned grid color and style for all plots
- Added constants for grid / label / tick colors and sizes
- Added constants for selection colors
- Added sorting icons
- Added vis hooks to the hexbin plot
- Vis hooks are also exported from visyn_core and can be used anywhere
- Vis hooks: zoom, selection, panning, (all from d3)
- Add warning for 'unsupported' vis type (e.g., vis types that no longer exist)

Open todos:
- [x] The barchart does not hide the data when clicking on the legend item (@dvdanielamoitzi)
  --> Legend click is now disabled
- [x] Barchart: first click on sort icon should sort it ascending (currently it's sorted descending) (@dvdanielamoitzi)
- [x] Heatmap: sorting icons
- [x] Correlation plot: label alignment
- [ ] Consolidate legend toggle (Make toggle available in all legends) (-> create new ticket @dvchristianbors)
- [x] Correlation plot: borders missing for edge labels (@dvdanielamoitzi)
- [ ] Consider label lengths when rotating labels in barchart (@dvdanielamoitzi)
  --> The rotation is currently working fine, but not considering the label length as suggested by @dvmoritzschoefl as we don't have the scales yet where we set the plot margins so we don't know the lenght of the longest label
- [x] check null values in barplot legend (it should be unknown) (@dvdanielamoitzi)

### CODE REVIEW

File: src/vis/bar/barComponents/XAxis.tsx [hunk 1]
```diff
@@ -45,27 +46,24 @@ export function XAxis({
...
 
       {ticks.map(({ value, offset }) => (
         <g key={value} transform={`translate(${offset}, ${vertPosition})`}>
-          <line y2="6" stroke="currentColor" />
           {showLines ? <line y2={`${-(yRange[1] - yRange[0])}`} stroke="lightgray" /> : null}
           <foreignObject x={0 - tickWidth / 2} y={10} width={tickWidth} height={20}>
             <Center>
               <Tooltip withinPortal label={value}>
-                <Text px={2} size={rem('10px')} style={{ textAlign: 'center', textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap' }}>
+                <Text c="gray.6" px={2} size={rem('10px')} style={{ textAlign: 'center', textOverflow: 'ellipsis', overflow: 'hidden', whiteSpace: 'nowrap' }}>
```
Comment (dvdanielamoitzi) [inline src/vis/bar/barComponents/XAxis.tsx:66]: The color constant should be used here @dvdanielamoitzi

File: src/vis/bar/barComponents/YAxis.tsx [hunk 2]
```diff
@@ -78,7 +71,7 @@ export function YAxis({
           >
             <foreignObject width={labelSpacing} height={20}>
               <Group style={{ width: '100%', height: '100%' }} justify="right">
-                <Text truncate size={rem('10px')}>
+                <Text c="gray.6" truncate size={rem('10px')}>
```
Comment (dvdanielamoitzi) [inline src/vis/bar/barComponents/YAxis.tsx:74]: The color constant should be used here @dvdanielamoitzi

**Comment by dvdanielamoitzi** [comment]
feedback by @dg-datavisyn :

- [ ] click on legend is not doing anything in the scatterplot
- [ ] hexbinplot: use selection color to selected points without categories
- [ ] barchart. NO color: selection = orange 
  - when categories: selection = opacity
- [ ] sankey: color selected orange

**Comment by dvdanielamoitzi** [comment]
Feedback from @dvchristianbors :
- [ ] the hover selection in the heatmap stays
- [ ] hexbinplot: the legend is incorrect
- [ ] the vis header does not show up anymore in Ordino
- [ ] remove the "test"
- [ ] vis type select: re-selecting the current option results in an empty plot

File: src/vis/heatmap/HeatmapVisSidebar.tsx [hunk 3]
```diff
@@ -17,7 +17,11 @@ export function HeatmapVisSidebar({
   return (
     <>
       <MultiSelect
-        callback={(catColumnsSelected: ColumnInfo[]) => setConfig({ ...config, catColumnsSelected })}
+        callback={(catColumnsSelected: ColumnInfo[]) => {
+          // For the heatmap, only 2 columns can be selected at once
+          const catColumns = catColumnsSelected.length > 2 ? catColumnsSelected.slice(catColumnsSelected.length - 2) : catColumnsSelected;
```
Comment (thinkh) [inline src/vis/heatmap/HeatmapVisSidebar.tsx:22]: Currently, you are using always the last two category columns, which could be written like this:


```suggestion
          // For the heatmap, only 2 columns can be selected at once
          const catColumns = catColumnsSelected.length > 2 ? catColumnsSelected.slice(-2) : catColumnsSelected;
```

If you would like to use the first two categories, it must be `catColumnsSelected.slice(2)` instead.

File: src/vis/vishooks/math/m4.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+// Scale components
+export const sxi = 0;
+export const syi = 5;
+export const szi = 10;
+
+/**
+ * Returns a new identity matrix. If out is specified, the result is copied into out.
+ */
+export function I(out: Matrix4x4 = new Float32Array(16)): Matrix4x4 {
```
Comment (thinkh) [inline src/vis/vishooks/math/m4.ts:44]: Please rename the function to something more meaningful.

File: src/vis/vishooks/math/v3.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable no-param-reassign */
+import { Vector3 } from './interfaces';
+
+export function I(out = new Float32Array(3)): Vector3 {
```
Comment (thinkh) [inline src/vis/vishooks/math/v3.ts:4]: Please rename the function to something more meaningful.

File: src/vis/vishooks/math/v3.ts
Comment (thinkh) [inline src/vis/vishooks/math/v3.ts:1]: Please rename the file to something more meaningful (e.g., vector3)?

File: src/vis/vishooks/math/m4.ts
Comment (thinkh) [inline src/vis/vishooks/math/m4.ts:1]: Please rename the file to something more meaningful (e.g., matrix4 or matrix4x4)?

File: src/vis/vishooks/vite-env.d.ts
Comment (thinkh) [inline src/vis/vishooks/vite-env.d.ts:1]: Can we remove this file?

File: src/vis/LegendItem.tsx
Comment (thinkh) [inline src/vis/LegendItem.tsx:1]: I'd move this file to the vis/general folder. We also have the SortIcon.tsx there.

File: src/vis/utils.ts
Comment (thinkh) [inline src/vis/utils.ts:1]: Move to general folder, too?

File: src/vis/violin/utils.ts [hunk 6]
```diff
@@ -88,6 +88,7 @@ export async function createViolinTraces(
           xaxis: config.separation === EViolinSeparationMode.GROUP || plotCounter === 1 ? 'x' : `x${plotCounter}`,
           yaxis: config.separation === EViolinSeparationMode.GROUP || plotCounter === 1 ? 'y' : `y${plotCounter}`,
           marker: {
+            // color: selectedList.length !== 0 && numCurr.resolvedValues.find((val) => selectedMap[val.id]) ? selectionColorDark : '#878E95',
```
Comment (thinkh) [inline src/vis/violin/utils.ts:91]: Remove?

File: src/vis/scatter/ScatterVis.tsx [hunk 7]
```diff
@@ -205,9 +205,8 @@ export function ScatterVis({
       showlegend: showLegend,
       legend: {
         // @ts-ignore
-        itemclick: false,
-        itemdoubleclick: false,
-
+        // itemclick: false,
+        // itemdoubleclick: false,
```
Comment (thinkh) [inline src/vis/scatter/ScatterVis.tsx:209]: Can we remove these three lines all together?

File: src/vis/sankey/SankeyVis.tsx [hunk 8]
```diff
@@ -136,8 +141,8 @@ function isNodeSelected(selection: Set<string>, inverseLookup: Array<string>) {
 
 function generatePlotly(data, optimisedSelection: Set<string>, theme: MantineTheme) {
   // @TODO @MORITZ
```
Comment (thinkh) [inline src/vis/sankey/SankeyVis.tsx:143]: Is this TODO still valid? Please specify or remove.

File: src/vis/heatmap/HeatmapVisSidebar.tsx (hunk 3 above)
Comment (dvdanielamoitzi) [reply src/vis/heatmap/HeatmapVisSidebar.tsx:22]: we'll do that in a follow-up pr

**Review by thinkh — APPROVED** [review]
@dvdanielamoitzi Thanks for addressing the latest fixes. The build is passing now and can be merged.

**Pull request merged**